NOTES.WIN

```diff
@@ -2,29 +2,12 @@
  NOTES FOR THE WINDOWS PLATFORMS
  ===============================
 
- Windows targets can be classified as "native", ones that use Windows API
- directly, and "hosted" which rely on POSIX-compatible layer. "Native"
- targets are VC-* (where "VC" stems from abbreviating Microsoft Visual C
- compiler) and mingw[64]. "Hosted" platforms are Cygwin and MSYS[2]. Even
- though the latter is not directly supported by OpenSSL Team, it's #1
- popular choice for building MinGW targets. In the nutshell MinGW builds
- are always cross-compiled. On Linux and Cygwin they look exactly as such
- and require --cross-compile-prefix option. While on MSYS[2] it's solved
- rather by placing gcc that produces "MinGW binary" code 1st on $PATH.
- This is customarily source of confusion. "Hosted" applications "live" in
- emulated file system name space with POSIX-y root, mount points, /dev
- and even /proc. Confusion is intensified by the fact that MSYS2 shell
- (or rather emulated execve(2) call) examines the binary it's about to
- start, and if it's found *not* to be linked with MSYS2 POSIX-y thing,
- command line arguments that look like file names get translated from
- emulated name space to "native". For example '/c/some/where' becomes
- 'c:\some\where', '/dev/null' - 'nul'. This creates an illusion that
- there is no difference between MSYS2 shell and "MinGW binary", but
- there is. Just keep in mind that "MinGW binary" "experiences" Windows
- system in exactly same way as one produced by VC, and in its essence
- is indistinguishable from the latter. (Which by the way is why
- it's referred to in quotes here, as "MinGW binary", it's just as
- "native" as it can get.)
+ Windows targets can be classified as "native", ones that use the
+ Windows API directly, and "hosted" which rely on POSIX-compatible layer.
+ "Native" targets are VC-* (where "VC" stems from Microsoft Visual C compiler)
+ and mingw[64]. "Hosted" platforms are Cygwin and MSYS[2].
+ Even though the latter is not directly supported by OpenSSL Team,
+ it's #1 popular choice for building MinGW targets.
```

Comment:
This text is actually quite confusing - although I understand what the original author was trying to say. Part of the confusion comes from the fact that you can build "native" targets (via mingw), using a "hosted" build environment (msys2). I think we can simplify some of this. How about:

> OpenSSL can be built in a number of different ways on the Windows platform. There are two main types: "native" and "hosted". A native build of OpenSSL uses the Windows APIs directly. A hosted build of OpenSSL relies on an external compatibility layer.
> 
> To build a native OpenSSL you have two build environment options. You can either use:
> 
> 1) a Visual C++ based environment
> or
> 2) a MinGW based environment
> 
> To build a "Hosted" OpenSSL you can use Cygwin.

Reply:
> This text is actually quite confusing

I agree. Thanks @mattcaswell for your clarifications,
which were of helped also to me to understand the various possibilities better.
I've taken over your suggestion for improved structuring and extended it.
This was more work than expected, but I think it was worth while.